wcfsetup/install/files/lib/system/package/PackageInstallationDispatcher.class.php

```diff
@@ -396,6 +396,10 @@ public function getArchive() {
 	protected function installPackage(array $nodeData) {
 		$installationStep = new PackageInstallationStep();
 		
+		if ($this->getAction() === 'update' && $this->getPackageName() === 'com.woltlab.wcf') {
```

Comment:
This doesn't work, because its check accomplishes nothing. `installPackage()` is the first node in the installation process, but the last one for an update. We're looking to validate the file permissions *before* the Core update takes place.